In [1]:
# Dependencies and Setup
import matplotlib.pyplot as plt
import pandas as pd
import scipy.stats as st
import numpy as np
import seaborn as sns


In [2]:
# Import the data/csv file & preview
df = pd.read_csv("cleaned_data.csv")
df


Unnamed: 0  unit_id check_in_date check_out_date reservation_type  \
0               0    40630      1/1/2018       1/3/2018            guest   
1               1    40630      1/3/2018       1/7/2018            guest   
2               2    40630     1/11/2018      1/15/2018            guest   
3               3    40630     1/15/2018      1/18/2018            owner   
4               4    40630     1/20/2018      1/21/2018            guest   
...           ...      ...           ...            ...              ...   
72451       92536    21239      3/9/2019      3/13/2019            guest   
72452       92538    21239     3/21/2019      3/24/2019            guest   
72453       92540    21239      4/9/2019      4/12/2019            guest   
72454       92541    21239     4/12/2019      4/14/2019            guest   
72455       92542    21239     4/14/2019      4/16/2019            guest   

             reservation_source_type reservation_created_date  adult_guests  \
0                            unknown                 1/1/2018             2   
1                            unknown                 1/3/2018             2   
2                            unknown                9/29/2017             2   
3                            unknown                1/15/2018             2   
4                            unknown                1/17/2018             2   
...                              ...                      ...           ...   
72451  3rd Party Channel Reservation                2/17/2019             4   
72452  3rd Party Channel Reservation                3/20/2019             4   
72453  3rd Party Channel Reservation                 4/8/2019             2   
72454                         Vacasa                 4/8/2019             2   
72455  3rd Party Channel Reservation                 4/8/2019             6   

       child_guests  pets  total_paid  city_name   zip5 state_name_short  \
0                 0     0        1930       Vail  81657               CO   
1                 0     0        2969       Vail  81657               CO   
2                 0     0       12913       Vail  81657               CO   
3                 0     0          60       Vail  81657               CO   
4                 0     0         515       Vail  81657               CO   
...             ...   ...         ...        ...    ...              ...   
72451             0     0        1300  Galveston  77554               TX   
72452             0     0         661  Galveston  77554               TX   
72453             0     0         340  Galveston  77554               TX   
72454             1     0         245  Galveston  77554               TX   
72455             0     0         287  Galveston  77554               TX   

          region_name_short  
0       Vail / Beaver Creek  
1       Vail / Beaver Creek  
2       Vail / Beaver Creek  
3       Vail / Beaver Creek  
4       Vail / Beaver Creek  
...                     ...  
72451  The Texas Gulf Coast  
72452  The Texas Gulf Coast  
72453  The Texas Gulf Coast  
72454  The Texas Gulf Coast  
72455  The Texas Gulf Coast  

[72456 rows x 15 columns]

In [3]:
# Find all reservations 
df['unit_id'].value_counts()

9131     157
11033    146
6309     145
17653    143
40618    143
        ... 
31087      1
31237      1
9107       1
30653      1
17594      1
Name: unit_id, Length: 2125, dtype: int64

In [4]:
# Get value count of # of children guests 
df['child_guests'].value_counts().describe()

count       22.000000
mean      3293.454545
std      11432.007692
min          2.000000
25%          6.000000
50%         59.500000
75%        692.500000
max      53841.000000
Name: child_guests, dtype: float64

In [5]:
# Divide # of children (max = 53841) / # of reservations (72456)
np.true_divide(53841, 72456)

0.7430854587611792

#### Cleaned data- % of reservations had children = 74.3% 

In [6]:
# Get count of reservations at Vail / Beaver Creek
df['region_name_short'].value_counts()

The Texas Gulf Coast    26351
Vail / Beaver Creek     25732
Delaware                20373
Name: region_name_short, dtype: int64

#### Delaware: 20,373 reservations

#### The Texas Gulf Coast: 26,351 reservations

#### Vail / Beaver Creek: 25,732 reservations

In [7]:
test = df.filter(['unit_id','region_name_short'])
test

unit_id     region_name_short
0        40630   Vail / Beaver Creek
1        40630   Vail / Beaver Creek
2        40630   Vail / Beaver Creek
3        40630   Vail / Beaver Creek
4        40630   Vail / Beaver Creek
...        ...                   ...
72451    21239  The Texas Gulf Coast
72452    21239  The Texas Gulf Coast
72453    21239  The Texas Gulf Coast
72454    21239  The Texas Gulf Coast
72455    21239  The Texas Gulf Coast

[72456 rows x 2 columns]

In [8]:
test['total']=test.groupby(by='unit_id')['unit_id'].transform('count')
test

unit_id     region_name_short  total
0        40630   Vail / Beaver Creek     86
1        40630   Vail / Beaver Creek     86
2        40630   Vail / Beaver Creek     86
3        40630   Vail / Beaver Creek     86
4        40630   Vail / Beaver Creek     86
...        ...                   ...    ...
72451    21239  The Texas Gulf Coast      6
72452    21239  The Texas Gulf Coast      6
72453    21239  The Texas Gulf Coast      6
72454    21239  The Texas Gulf Coast      6
72455    21239  The Texas Gulf Coast      6

[72456 rows x 3 columns]

In [9]:
# Reservations & Unit Countsd
test.groupby(by="region_name_short")["unit_id"].nunique()

region_name_short
Delaware                939
The Texas Gulf Coast    512
Vail / Beaver Creek     674
Name: unit_id, dtype: int64

In [10]:
# Find all reservations 
df['unit_id'].value_counts()

9131     157
11033    146
6309     145
17653    143
40618    143
        ... 
31087      1
31237      1
9107       1
30653      1
17594      1
Name: unit_id, Length: 2125, dtype: int64

#### Number of units that had a reservation = 2,125 units
#### Number of total reservation in cleaned data = 72,456

In [11]:
# Divide # of children (max = 53841) / # of reservations (72456)
np.true_divide(72456, 53841)

1.3457402351367918

In [12]:
df['child_guests'].value_counts()

0     53841
2      6887
1      4746
3      3026
4      2071
5       777
6       439
8       203
7       173
10       82
12       64
9        55
11       29
15       22
14       12
13        9
18        5
16        4
99        4
19        3
20        2
17        2
Name: child_guests, dtype: int64

#### 309 is total # on left column added up. (not sure how to do this, need to research more).


In [13]:
# divide total count (309) by total reservations 
np.true_divide(309,72456)

0.004264657171248758

#### Average number of children on a reservation = 4